extensions/wikidata/module/scripts/dialogs/wikibase-dialog.js

```diff
@@ -0,0 +1,128 @@
+  elmts.addButton.text($.i18n("wikibase-management/add-wikibase"));
+
+  WikibaseDialog.populateDialog();
+
+  let level = DialogSystem.showDialog(frame);
+
+  elmts.closeButton.click(function () {
+    DialogSystem.dismissUntil(level - 1);
+  });
+
+  elmts.addButton.click(function () {
+    WikibaseDialog.addWikibaseManifest();
+  });
+};
+
+WikibaseDialog.populateDialog = function () {
+  let wikibases = WikibaseManager.getAllWikibases();
+
+  WikibaseDialog.elmts.wikibaseList.empty();
+  for (let wikibaseName in wikibases) {
+    if (wikibases.hasOwnProperty(wikibaseName)) {
+      let item = "<tr onclick=\"WikibaseDialog.selectWikibase('" + wikibaseName + "')\">";
+      item += "<td>" + wikibaseName + "</td>";
+      if (wikibaseName.toLowerCase() === WikibaseManager.getSelectedWikibaseName().toLowerCase()) {
+        item += "<td><a class=\"wikibase-dialog-selector-remove wikibase-selected\" onclick=\"void(0)\"></a></td>";
+      } else {
+        item += "<td><a class=\"wikibase-dialog-selector-remove\" onclick=\"WikibaseDialog.removeWikibase(event, '" + wikibaseName + "')\"></a></td>";
+      }
+      item += "</tr>";
+      WikibaseDialog.elmts.wikibaseList.append(item);
+    }
+  }
+};
+
+WikibaseDialog.selectWikibase = function (wikibaseName) {
+  if (wikibaseName !== WikibaseManager.getSelectedWikibaseName()) {
+    WikibaseManager.selectWikibase(wikibaseName);
+    WikibaseDialog.elmts.currentSelectedWikibase.html($.i18n("wikibase-management/current-selected-wikibase",
+        WikibaseManager.getSelectedWikibaseMainPage(), WikibaseManager.getSelectedWikibaseName()));
+    WikibaseDialog.populateDialog();
+    SchemaAlignment.onWikibaseChange();
+  }
+};
+
+WikibaseDialog.removeWikibase = function (e, wikibaseName) {
+  e.stopPropagation(); // must stop, otherwise the removed Wikibase will be selected
+  WikibaseManager.removeWikibase(wikibaseName);
+  WikibaseDialog.populateDialog();
+};
+
+
+WikibaseDialog.addWikibaseManifest = function () {
+  const frame = $(DOM.loadHTML("wikidata", "scripts/dialogs/add-wikibase-dialog.html"));
+  const elmts = DOM.bind(frame);
+  elmts.dialogHeader.text($.i18n("wikibase-addition/dialog-header"));
+  elmts.explainAddManifest.text($.i18n("wikibase-addition/explain-add-manifest"));
+  elmts.explainAddManifestViaURL.text($.i18n("wikibase-addition/explain-add-manifest-via-url"));
+  elmts.explainPasteManifest.html($.i18n("wikibase-addition/explain-paste-manifest"));
+  elmts.cancelButton.text($.i18n("wikibase-addition/cancel"));
+  elmts.addButton.text($.i18n("wikibase-addition/add-wikibase"));
+  elmts.invalidManifest.hide();
+  elmts.invalidManifest.text($.i18n("wikibase-addition/invalid-manifest"));
+
+  let level = DialogSystem.showDialog(frame);
+
+  elmts.cancelButton.click(function () {
+    DialogSystem.dismissUntil(level - 1);
+  });
+
+  elmts.addButton.click(function () {
+    let addManifest = function (manifest) {
+      if (!manifest) {
+        alert("manifest is empty");
```

Comment:
Can these error messages be internationalized?

Reply:
The work on checking the format of the manifest is not finished yet, I'll add i18n support when this part is done.

Reply:
Thanks for letting me know that it will be addressed. I'm trying to strike a balance between commenting constantly and ignoring you. :) I'm also concerned that the PR is becoming enormous and will be completely unreviewable if I don't do it incrementally.